File: BackEnd/java/ZaloMini/src/main/java/com/devmam/zalomini/entities/MessageReaction.java
package com.devmam.zalomini.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.Instant;
import java.util.Date;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity(name = "MessageReaction")
@Table(name = "message_reactions")
public class MessageReaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "message_id", nullable = false)
    private Message message;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Lob
    @Column(name = "reaction_type", nullable = false)
    private String reactionType;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "created_at")
    @CreationTimestamp
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

}

File: BackEnd/Zalo_mini/Zalo_mini/Models/message_reaction.cs
using System;
using System.Collections.Generic;

namespace Zalo_mini.Models
{
    public partial class message_reaction
    {
        public long id { get; set; }
        public long message_id { get; set; }
        public long user_id { get; set; }
        public string reaction_type { get; set; } = null!;
        public DateTime? created_at { get; set; }

        public virtual message message { get; set; } = null!;
        public virtual user user { get; set; } = null!;
    }
}
